### PR TITLE
Implemented LeviCivita Derivative and Added Test

### PR DESCRIPTION
**Pull request by ShikharJ**
Relevant: #814 
Please review and suggest improvements.

### CODE REVIEW

File: symengine/derivative.cpp [hunk 1]
```diff
@@ -670,6 +669,18 @@ class DiffImplementation
...
         }
     }
+
+    static RCP<const Basic> diff(const LeviCivita &self,
+                                 const RCP<const Symbol> &x)
+    {
+        for (auto &a : self.get_args()) {
+            auto b = a->diff(x);
+            if (neq(*b, *zero)) {
+                return Derivative::create(self.rcp_from_this(), {x});
```
Comment (isuruf) [inline symengine/derivative.cpp]: This should be done in the same way as `FunctionSymbol`.

Comment (isuruf) [inline symengine/derivative.cpp]: See SymPy output,
```

In [13]: LeviCivita(x, y).diff(x)
Out[13]: Derivative(LeviCivita(x, y), x)

In [14]: LeviCivita(x**2, y).diff(x)
Out[14]: 2*x*Subs(Derivative(LeviCivita(_xi_1, y), _xi_1), (_xi_1,), (x**2,))

In [15]: LeviCivita(x**2, 2*x).diff(x)
Out[15]: 2*x*Subs(Derivative(LeviCivita(_xi_1, 2*x), _xi_1), (_xi_1,), (x**2,)) + 2*Subs(Derivative(LeviCivita(x**2, _xi_2), _xi_2), (_xi_2,), (2*x,))
```

File: symengine/derivative.cpp [hunk 2]
```diff
@@ -670,6 +669,40 @@ class DiffImplementation
             }
         }
     }
+
+    static RCP<const Basic> diff(const LeviCivita &self,
+                                 const RCP<const Symbol> &x)
+    {
+        RCP<const Basic> diff = zero;
```
Comment (isuruf) [inline symengine/derivative.cpp]: Can you change this, so that there is no code duplication?

File: symengine/derivative.cpp [hunk 3]
```diff
@@ -670,6 +641,46 @@ class DiffImplementation
...
     }
+
+    static RCP<const Basic> diff(const LeviCivita &self,
+                                 const RCP<const Symbol> &x)
+    {
+        return diff0(self, x);
+    }
+
+    template <typename T>
+    static RCP<const Basic> diff0(const T &self, const RCP<const Symbol> &x)
```
Comment (isuruf) [inline symengine/derivative.cpp]: How about changing this to `static RCP<const Basic> diff(const Function &self, const RCP<const Symbol> &x)` ?

Comment (ShikharJ) [inline symengine/derivative.cpp]: @isuruf I initially tried to do that, by first trying `Basic` and then `Function`, but always got an error. These classes do not have the function `create`.
```
/home/shikhar/symengine/symengine/derivative.cpp: In static member function ‘static Teuchos::RCP<const SymEngine::Basic> SymEngine::DiffImplementation::diff(const SymEngine::Function&, const Teuchos::RCP<const SymEngine::Symbol>&)’:
/home/shikhar/symengine/symengine/derivative.cpp:679:56: error: ‘const class SymEngine::Function’ has no member named ‘create’
                                Derivative::create(self.create(v), {v[i]}), m)));
```

Comment (isuruf) [inline symengine/derivative.cpp]: Use `MultiArgFunction` which has `create`

File: symengine/tests/basic/test_functions.cpp [hunk 4]
```diff
@@ -970,30 +971,30 @@ TEST_CASE("Derivative: functions", "[functions]")
...
     f = function_symbol("f", {y, add(x, y)});
     r1 = f->diff(x);
     std::cout << *f << " " << *r1 << std::endl;
-    r2 = Derivative::create(function_symbol("f", {y, _x}), {_x});
-    r2 = Subs::create(r2, {{_x, add(y, x)}});
+    r2 = Derivative::create(function_symbol("f", {y, _xi_2}), {_xi_2});
+    r2 = Subs::create(r2, {{_xi_2, add(y, x)}});
     REQUIRE(eq(*r1, *r2));
 
     r1 = function_symbol("f", add(_x, x))->diff(_x);
```
Comment (isuruf) [inline symengine/tests/basic/test_functions.cpp]: change `_x` to `_xi_1` to keep the purpose of this test.

**Comment by isuruf** [comment]
It seems `FunctionSymbol` is not a `MultiArgFunction`, which it should be. Can you change the parent of `FunctionSymbol` to be `MultiArgFunction`?

File: symengine/functions.h [hunk 5]
```diff
@@ -128,8 +128,9 @@ class MultiArgFunction : public Function
     vec_basic arg_;
 
 public:
-    //! Constructor
+    //! Constructors
     MultiArgFunction(const vec_basic &arg) : arg_{arg} {};
+    MultiArgFunction(const RCP<const Basic> &arg) : arg_{{arg}} {};
```
Comment (isuruf) [inline symengine/functions.h]: Remove this. You can have a similar constructor FunctionSymbol

File: symengine/functions.cpp [hunk 6]
```diff
@@ -1258,14 +1258,14 @@ RCP<const Basic> lambertw(const RCP<const Basic> &arg)
 }
 
 FunctionSymbol::FunctionSymbol(std::string name, const RCP<const Basic> &arg)
-    : name_{name}, arg_{{arg}}
+    : MultiArgFunction(std::move(arg)), name_{name}, arg_{{arg}}
```
Comment (isuruf) [inline symengine/functions.cpp]: You are storing `arg` in both. You need to remove `arg_`

File: symengine/functions.cpp [hunk 7]
```diff
@@ -1396,6 +1397,17 @@ bool Derivative::is_canonical(const RCP<const Basic> &arg,
             }
         }
         return found;
+    } else if (is_a<LeviCivita>(*arg)) {
```
Comment (isuruf) [inline symengine/functions.cpp]: Code duplication. Use the same function for `FunctionSymbol`

**Comment by isuruf** [comment]
Thanks for the PR

**Pull request merged**